File: codesignal/arcade/2.core/6.labyrinth of nested loops/2.isSumOfConsecutive2.cpp
// Find the number of ways to express n as sum of some (at least two) consecutive positive integers.

class Solution {
public:
  int isSumOfConsecutive2(int n) {
    int count = 0;
    for (int i = 1; i < n; i++) {
      int sum = 0;
      for (int j = i; j < n; j++) {
        sum += j;
        if (sum == n) {
          count++;
          break;
        }
        if (sum > n) break;
      }
    }
    return count;
  }
};

File: codesignal/arcade/2.core/6.labyrinth of nested loops/2.isSumOfConsecutive2.py
# Find the number of ways to express n as sum of some (at least two) consecutive positive integers.

class Solution:
    def isSumOfConsecutive2(self, n):
        count = 0
        for i in range(1, n):
            sum = 0
            for j in range(i, n):
                sum += j
                if sum == n:
                    count += 1
                    break
                elif sum > n:
                    break
        return count